command/agent/config_parse.go

```diff
@@ -406,10 +400,17 @@ func parseVaults(c *Config, list *ast.ObjectList) error {
 		if v.Name == "" {
 			v.Name = structs.VaultDefaultCluster
 		}
-		if exist, ok := c.Vaults[v.Name]; ok {
-			c.Vaults[v.Name] = exist.Merge(v)
-		} else {
-			c.Vaults[v.Name] = v
+
+		var vaultFound bool
+		for i, exist := range c.Vaults {
+			if exist.Name == v.Name {
+				c.Vaults[i] = exist.Merge(v)
+				vaultFound = true
+				break
```

Comment:
Should we actually break here? This loop seems to be merging all `vault` blocks with the same name (kind of an unfortunate behaviour I feel, but we can't really change it at this point 😅)

By breaking early we're only merging once, so a config with 3+ `vault` blocks with the same name (oof...) will only contain the config from the first two.

Reply:
I'm pretty sure this is ok because `c.Vaults` will always start the `parseVaults` method empty. And this loop gets called for each `vault` block. So suppose we have the following silly case:

```hcl
vault { name = "foo" }
vault { name = "bar" }
vault { name = "foo" }
vault { name = "foo" }
```

The first block will be `c.Vaults[0]`. The second block will be `c.Vaults[1]`. The third block will get merged onto `c.Vaults[0]`. The fourth block (the 3rd "foo" block) will get merged onto `c.Vaults[0]`.